packages/react/src/components/MultiSelect/__tests__/MultiSelect-test.js

```diff
@@ -3,380 +3,443 @@
-      ).toBe(1);
-      simulateArrowDown(wrapper);
-      wrapper.find(`.${prefix}--list-box__field`).simulate('keydown', {
-        key: 'Enter',
-      });
-      expect(
-        wrapper.find(`.${prefix}--list-box__menu-item--active`).length
-      ).toBe(0);
-    });
+    it('should support an invalid state with invalidText that describes the field', () => {
+      const items = generateItems(4, generateGenericItem);
+      const label = 'test-label';
 
-    it('should allow a user to click on the clear icon to clear all selected items', () => {
-      const wrapper = mount(
-        <MultiSelect id="test-multiselect" {...mockProps} />
+      const { container } = render(
+        <MultiSelect
+          id="custom-id"
+          invalid
+          invalidText="Fool of a Took!"
+          label={label}
+          items={items}
+        />
       );
-      openMenu(wrapper);
-      wrapper
-        .find(`.${prefix}--list-box__menu-item`)
-        .at(0)
-        .simulate('click');
-      wrapper.find(`.${prefix}--list-box__selection`).simulate('click');
-      expect(wrapper.find('Selection').instance().state.selectedItems).toEqual(
-        []
+
+      expect(getByText(container, 'Fool of a Took!')).toBeTruthy();
+
+      expect(document.querySelector('[data-invalid="true"')).toBeInstanceOf(
+        HTMLElement
       );
     });
 
-    it('should allow a user to focus the clear icon and hit enter to clear all selected items', () => {
-      const wrapper = mount(
-        <MultiSelect id="test-multiselect" {...mockProps} />
+    it('should support different feedback modes with selectionFeedback', () => {
+      const items = generateItems(4, generateGenericItem);
+      const label = 'test-label';
+      // eslint-disable-next-line no-unused-vars
+      const [firstItem, secondItem, thirdItem, fourthItem] = items;
+      const { container } = render(
+        <MultiSelect
+          id="custom-id"
+          selectionFeedback="top"
+          label={label}
+          items={items}
+        />
       );
-      openMenu(wrapper);
-      wrapper
-        .find(`.${prefix}--list-box__menu-item`)
-        .at(0)
-        .simulate('click');
-      wrapper.find(`.${prefix}--list-box__selection`).simulate('keydown', {
-        keyCode: 13,
-      });
-      expect(wrapper.find('Selection').instance().state.selectedItems).toEqual(
-        []
+
+      // click the label to open the multiselect options menu
+      const labelNode = getByText(container, label);
+      Simulate.click(labelNode);
+
+      // click the third option down in the list
+      const itemNode = getByText(container, thirdItem.label);
+      Simulate.click(itemNode);
+
+      // get an array of all the options
+      const optionsArray = Array.from(
+        document.querySelectorAll('[role="option"]')
       );
+
+      // the first option in the list to the the former third option in the list
+      expect(optionsArray[0].title).toMatch('Item 2');
```

Comment:
I think we might need to use `toBe` for the direct check as `toMatch` is a partial match checker. What I mean by that is that the text `'Item 2'` and `'Item 2 plus'` both would match `'Item 2'`

```suggestion
      expect(optionsArray[0].title).toBe('Item 2');
```

Similarly, you might be able to do:

```suggestion
      expect(optionsArray[0]).toHaveProperty('title', 'Item 2');
```